eth/beacon/state_machines/base.py

```diff
@@ -368,33 +373,108 @@ def compute_per_block_transition(cls,
     def compute_cycle_transitions(
             cls,
             crystallized_state: CrystallizedState,
-            active_state: ActiveState) -> Tuple[CrystallizedState, ActiveState]:
-        # TODO: it's a stub
+            active_state: ActiveState,
+            block: BaseBeaconBlock) -> Tuple[CrystallizedState, ActiveState]:
+        """
+        Compute the cycle transitions and return processed CrystallizedState and ActiveState.
+        """
+        while block.slot_number >= crystallized_state.last_state_recalc + cls.config.CYCLE_LENGTH:
```

Comment:
This `while` loop will never end with the current stub of `compute_per_cycle_transition`.

Consider updating the stub to at least update `new_crystallized_state.last_state_recalc += crystallized_state.last_state_recalc + CYCLE_LENGTH`

Reply:
and add a sanity check test that we can go through a cycle transition without it hanging in the while loop

Reply:
**if** it is possible to determine up-front how many iterations this will take (which I recognize may not be possible) I think it might be prudent to do that so that we can write this loop in a manner that is guaranteed to terminate.

Alternatively, can we compute a reasonable upper bound and set the loop to only run that long.  The inner loop can use a `break` and then we can add an exception to the loop's `else` clause to prevent a possible infinite loop.

All of this assumes that the algorithms involved here are sufficiently complex that we won't be **really** sure this loop's exit condition will be met which currently seems to be the case.

Reply:
> if it is possible to determine up-front how many iterations this will take

I think yes, if no accident, it should be `(block.slot_number - crystallized_state.last_state_recalc) // cls.config.CYCLE_LENGTH` times since according to the spec, the `crystallized_state.last_state_recalc` will be updated to `crystallized_state.last_state_recalc + cls.config.CYCLE_LENGTH` at the end of `compute_per_cycle_transition` too.

Reply:
I see you pinged me for review but I don't see any of the suggested changes to this loop's structure.  Is that being done separately?